### PR TITLE
feat: profile-resolve command to generate resolved profile catalog

### LINKED ISSUE #1110: Provide new command to generate resolved profile catalog from profile
## Issue description / feature objectives
Several of the authoring tools rely on internally generating the resolved profile catalog from a profile, but we don't have a command simply to create that catalog on its own.

Suggest adding a new command, `resolve` that takes a profile as input and generates a json resolved profile catalog.

### PR DESCRIPTION
**Pull request by fsuits**
Signed-off-by: Frank Suits <frankst@au1.ibm.com>

## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [ ] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary
New feature profile-resolve will generate resolved profile catalog from profile
Option to use the [Assignment:] form (default) or substitute values in prose
Removed the redundant script to generate resolved profile catalog

closes #1110 

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

File: trestle/core/commands/author/profile.py [hunk 1]
```diff
@@ -428,3 +428,65 @@ def assemble_profile(
...
+            param_rep = ParameterRep.VALUE_OR_LABEL_OR_CHOICES if show_values else ParameterRep.ASSIGNMENT_FORM
+            catalog = ProfileResolver().get_resolved_profile_catalog(
+                trestle_root, profile_path, False, False, None, param_rep
+            )
+            ModelUtils.save_top_level_model(catalog, trestle_root, catalog_name, FileContentType.JSON)
+
+        except TrestleNotFoundError as e:
+            raise TrestleError(f'Profile {profile_path} not found, error {e}')
+        except TrestleError as e:
+            raise TrestleError(f'Error generating the resolved profile catalog: {e}')
```
Comment (enikonovad) [inline trestle/core/commands/author/profile.py:491]: No need to catch a generic TrestleError here, let it fall up

File: trestle/core/commands/author/profile.py [hunk 2]
```diff
@@ -428,3 +428,65 @@ def assemble_profile(
...
+        """Create resolved profile catalog from given profile.
+
+        Args:
+            trestle_root: Root directory of the trestle workspace
+            profile_path: Path of the profile json file
+            catalog_name: Name of the resolved profile catalog
+            show_values: If true, show values of parameters in prose rather than [Assignment:] form
+
+        Returns:
+            0 on success, 1 on error
```
Comment (enikonovad) [inline trestle/core/commands/author/profile.py:479]: returns 0 on success but raises exception on error

File: tests/trestle/core/commands/author/profile_test.py [hunk 3]
```diff
@@ -808,3 +808,21 @@ def generate_assemble(md_path: pathlib.Path) -> MarkdownNode:
...
     tree = generate_assemble(ac1_path)
     assert not tree.get_node_for_key('## Control this_should_appear_in_parts')
+
+
+@pytest.mark.parametrize('show_values', [True, False])
+def test_profile_resolve(tmp_trestle_dir: pathlib.Path, show_values: bool, monkeypatch: MonkeyPatch) -> None:
+    """Test profile resolve to create resolved profile catalog."""
+    test_utils.setup_for_multi_profile(tmp_trestle_dir, False, False)
+    cat_name = 'resolved_catalog'
+    command_profile_resolve = f'trestle author profile-resolve -n main_profile -o {cat_name}'
```
Comment (enikonovad) [inline tests/trestle/core/commands/author/profile_test.py:818]: More test scenarios would be beneficial, like non-existing profile or invalid profile

**Review by enikonovad — APPROVED** [review]
LGTM - some minor comments, please add docs

File: trestle/core/commands/author/profile.py (hunk 1 above)
Comment (fsuits) [reply trestle/core/commands/author/profile.py:491]: fixed

File: trestle/core/commands/author/profile.py (hunk 2 above)
Comment (fsuits) [reply trestle/core/commands/author/profile.py:479]: fixed

File: tests/trestle/core/commands/author/profile_test.py (hunk 3 above)
Comment (fsuits) [reply tests/trestle/core/commands/author/profile_test.py:818]: added two failure tests

**Pull request merged**